### PR TITLE
qu2cu

### PR DESCRIPTION
**Pull request by behdad**

### CODE REVIEW

**Comment by behdad** [comment]
Based on https://github.com/googlefonts/cu2qu/pull/29/files#diff-0ad9d3d850d1321d2e2abe961cabbef118134bb5c155ee65103968c8c2fe16acR274-R286

File: Lib/fontTools/qu2cu/qu2cu.py [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+
+
+@cython.locals(_1_3=cython.double, _2_3=cython.double)
+@cython.locals(
+    p0=cython.complex,
+    p1=cython.complex,
+    p2=cython.complex,
+    p1_2_3=cython.complex,
+)
+def elevate_quadratic(p0, p1, p2, _1_3=1 / 3, _2_3=2 / 3):
```
Comment (rsheeter) [inline Lib/fontTools/qu2cu/qu2cu.py:47]: Suggest adding a citation so those new to curves can learn something, e.g. https://pomax.github.io/bezierinfo/#reordering

**Comment by rsheeter** [comment]
I was unable to locate the tests. Now we have cu2qu and qu2cu you could round trip things, perhaps cherry-picking cases where that works out with nice #s.

**Comment by behdad** [comment]
> I was unable to locate the tests. Now we have cu2qu and qu2cu you could round trip things, perhaps cherry-picking cases where that works out with nice #s.

Yeah right now there's just a main that roundtrips. Will need tests indeed.

**Comment by behdad** [comment]
The algorithm is slow (`O(n^3)`). But for short segments that `cu2qu` generates it should not matter.

**Comment by behdad** [comment]
Later we can add `quadratics_to_curve` that takes multiple (joining) quadratic splines and converts them.

**Comment by behdad** [comment]
I think this is ready now. With tests.

**Comment by behdad** [comment]
I've rebased it on cubic-glyf branch, but that's only required for the cli, which is not tested. So can be rebased on main and merged.

**Comment by behdad** [comment]
> can be rebased on main

Done.

**Comment by behdad** [comment]
Okay I'm finally happy with this.

File: Lib/fontTools/qu2cu/qu2cu.py [hunk 2]
```diff
@@ -0,0 +1,327 @@
...
+
+
+def quadratics_to_curves(pp, tolerance=0.5, all_cubic=False):
+    is_complex = type(pp[0][0]) is complex
+    if not is_complex:
+        pp = [[complex(x, y) for (x, y) in p] for p in pp]
+
+    q = [pp[0][0]]
+    cost = 0
+    costs = [0]
```
Comment (anthrotype) [inline Lib/fontTools/qu2cu/qu2cu.py:173]: it's unclear what these `costs` are referring to or how they are used, can you please add a comment at least to help the reader?

**Review by anthrotype — APPROVED** [review]
LGTM, thank you very much for pushing this to the finishing line! 👏

**Pull request merged**